api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -172,8 +192,12 @@ public Builder setAttribute(String key, AttributeValue value) {
      * @return this Builder
      */
     public Builder setAttribute(String key, String value) {
+      AttributeValue v = stringAttributeValue(value);
+      if (doNotAdd(key, v)) {
```

Comment:
rather than repeating this logic over and over again, can we just call the setAttribute method with the attribute value above?